src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -715,33 +705,54 @@ const TransactionBuilderContextProvider = ({
       setTransactionBlockFieldValidationError,
       setCrossChainActionGasToken,
     }),
-    [],
+    []
   );
 
   const hideMenu = () => setShowMenu(false);
 
   const hasTransactionBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE
   );
 
   const hasKlimaBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE
   );
 
   const crossChainActionsInProcessing = useMemo(() => {
     if (!processingCrossChainActionIds?.length) return;
-    return dispatchedCrossChainActions?.filter(
-      (crossChainAction) => processingCrossChainActionIds.some((id) => id === crossChainAction.id),
+    return dispatchedCrossChainActions?.filter((crossChainAction) =>
+      processingCrossChainActionIds.some((id) => id === crossChainAction.id)
     );
   }, [processingCrossChainActionIds, dispatchedCrossChainActions]);
 
-  const hasProcessingUnsent = useMemo(() => !!getFirstCrossChainActionByStatus(
-    dispatchedCrossChainActions ?? [],
-    CROSS_CHAIN_ACTION_STATUS.UNSENT,
-  ), [dispatchedCrossChainActions]);
+  const hasProcessingUnsent = useMemo(
+    () => !!getFirstCrossChainActionByStatus(dispatchedCrossChainActions ?? [], CROSS_CHAIN_ACTION_STATUS.UNSENT),
+    [dispatchedCrossChainActions]
+  );
 
   const [showMulticallOptions, setShowMulticallOptions] = useState<string | null>(null);
 
+  // Mt Pelerin
+  const [deployingAccount, setDeployingAccount] = useState(false);
+
+  const onBuyClick = async (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    event.preventDefault();
+
+    const maticSdk = getSdkForChainId(CHAIN_ID.POLYGON);
+
+    if (!accountAddress || !maticSdk) return;
+
+    let account = await maticSdk.getAccount();
+    if (!account || account.address !== accountAddress) {
+      await maticSdk.computeContractAccount();
+      account = await maticSdk.getAccount();
+    }
```

Comment:
maybe add exception handler and alert modal in case it goes wrong to not leave user hanging